### DISCUSSION TITLE
singUp does not throw an error if a user already exists

### DISCUSSION THREAD

**Discussion by fayazara** [supabase/supabase#4105] [Questions] [answered]
I am trying to handle emails which already exists, but supabase just returns me a user object instead, how do I handle this?

Plus, the user object returned even says `authenticated` even if I have not confirmed my email, is there anyway to handle this the correct way?

```
{
    "id": "b7dd04ed-14ec-4a2d-b2c3-d1086d704619",
    "aud": "authenticated",
    "role": "authenticated",
    "email": "testfyz@datab.info",
    "phone": "",
    "confirmation_sent_at": "2021-11-27T14:40:15.368421705Z",
    "app_metadata": {
        "provider": "email",
        "providers": [
            "email"
        ]
    },
    "user_metadata": {
        "company_name": "My Company",
        "name": "Test User"
    },
    "identities": [],
    "created_at": "2021-11-27T14:40:15.365933Z",
    "updated_at": "2021-11-27T14:40:17.817411Z"
}
```

Using this hack for now, but it doesn't seem right way.
```
if (user.email_confirmed_at)
    throw "User already exists with the email id";
else throw "Please confirm your email address";
```

**Comment by kangmingtay** [comment, marked as answer]
Hey @fayazara, we return an obfuscated / fake user object if someone tries to sign up with an email / phone number that already exists. This is because we want to prevent leaking out of information on whether a user is registered or not from an attacker. If you want to check if a user exists, you can call the `supabase.auth.api.getUser(access_token)`. If a user obj is returned, then the user exists.

For more context on the rationale behind this, you can check out: https://github.com/supabase/supabase-js/issues/296#issuecomment-976200828

**Reply by lockinmarv** [reply]
Hey! Where would I get that access token from when I call `supabase.auth.signUp()`? Not sure but I think it would be in the `session` of the return value which is empty if email confirmation is active.
